roadmap.md

```diff
@@ -0,0 +1,54 @@
+# WebDriver BiDi roadmap
+
+## Real-world end-to-end user scenarios
+
+This document presents an overview of real-world end-to-end user scenarios we aim to enable via the WebDriver BiDi protocol. Each scenario requires one or more WebDriver BiDi commands and events to be specified, tested, and implemented across browser engines.
+
+### A. Extracting content
+
+This scenario loads a web page and uses script evaluation to extract content on the page (e.g. the headlines). In spec terms, this involves:
+
+- [x] [handling sessions](https://w3c.github.io/webdriver-bidi/#module-session)
+- [x] [creating a new `browsingContext`](https://w3c.github.io/webdriver-bidi/#command-browsingContext-create)
+- [x] [navigating to a URL](https://w3c.github.io/webdriver-bidi/#command-browsingContext-navigate)
+- [x] [evaluating JavaScript in the page context](https://w3c.github.io/webdriver-bidi/#command-script-evaluate)
+- [x] [serialization/deserialization of JavaScript values](https://w3c.github.io/webdriver-bidi/#data-types-protocolValue)
+- [ ] [sandboxed script execution](https://github.com/w3c/webdriver-bidi/issues/144)
```

Comment:
Although I agree sandboxed script exection is a high priority feature, I'd say the use case it blocks is more like "enable writing robust generic clients". Although running script directly in the context of the page is bad, for simple cases like implementing element selection it's almost always OK to just run `document.querySelector()` since it has no side effects. You only need to worry about sandboxing to deal with more edge case scenarios like pages which overwrite `querySelector` or where you're doing complex things that require storing significant browser-side state that should be invisible to the page.

Reply:
Agreed; sandboxing is only required in cases the web page is non-cooperative/hostile.

Do you have a preference:

- move sandboxing from this scenario to another one (if so, where?)
- keep sandboxing here, but clarify in the scenario description that it’s only required for robustness

Reply:
I'd add a scenario specifically about "generic clients" or something that contains this and botostrap scripts.

Reply:
Starting point:

```
### X. Enabling robust generic clients

This scenario enables end-to-end user scenarios for web pages whose environments aren’t guaranteed to be pristine (e.g. the page’s `document.querySelector` and `Object.prototype.*` have been overwritten), or scenarios that require running JavaScript before the page’s own scripts.

- [x] (everything from the previous milestone)
- [ ] [sandboxed script execution](https://github.com/w3c/webdriver-bidi/issues/144)
- [ ] [installing bootstrap scripts](https://github.com/w3c/webdriver-bidi/blob/master/proposals/bootstrap-scripts.md#record-navigation-performance)
```

Any suggestions on how to make this a proper end-to-end user scenario? It would be nice to have a specific scenario that can be turned into a runnable script, like all the other entries on the current roadmap draft.

Reply:
Well any of the content extraction ones, but you get to supply a URL rather than it being for a specfic URL where you happen to know that the injected scripts don't clash with the page-defined scripts. Or maybe a specific URL where they would clash, except for the sandboxing / use of bootstrap scripts to inject early.

Reply:
I wonder if this should actually be a e2e example or not. It's clearly something for the milestone list but not sure if it fits here given that we want to keep the list short.